File: simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/data/tables/TaskTable.java
package com.zhuinden.simplestackexamplemvvm.data.tables;

import android.content.ContentValues;
import android.database.Cursor;

import com.zhuinden.simplestackexamplemvvm.core.database.BaseTable;
import com.zhuinden.simplestackexamplemvvm.core.database.DatabaseManager;
import com.zhuinden.simplestackexamplemvvm.data.Task;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Zhuinden on 2017.07.26..
 */

@Singleton
public class TaskTable
        extends BaseTable
        implements DatabaseManager.Mapper<Task> {
    public static final String NAME = "task";

    public static final DatabaseManager.Fields ENTRY_ID = new DatabaseManager.Fields("entry_id", "text", "primary key");
    public static final DatabaseManager.Fields TITLE = new DatabaseManager.Fields("title", "text", "");
    public static final DatabaseManager.Fields DESCRIPTION = new DatabaseManager.Fields("description", "text", "");
    // @formatter:off
    public static final DatabaseManager.Fields COMPLETED = new DatabaseManager.Fields("completed", "integer", ""); // boolean
    // @formatter:on

    public static final String $ENTRY_ID = NAME + "." + ENTRY_ID.getFieldName();
    public static final String $TITLE = NAME + "." + TITLE.getFieldName();
    public static final String $DESCRIPTION = NAME + "." + DESCRIPTION.getFieldName();
    public static final String $COMPLETED = NAME + "." + COMPLETED.getFieldName();

    @Inject
    public TaskTable() {
    }

    @Override
    public String getTableName() {
        return NAME;
    }

    @Override
    public DatabaseManager.Fields[] getFields() {
        return new DatabaseManager.Fields[]{ENTRY_ID, TITLE, DESCRIPTION, COMPLETED};
    }

    @Override
    public String getId(Object element) {
        return ((Task)element).id();
    }

    @Override
    public String getIdFieldName() {
        return ENTRY_ID.getFieldName();
    }

    @Override
    public Task from(Cursor cursor) {
        return Task.newBuilder()
                .setId(cursor.getString(cursor.getColumnIndexOrThrow(ENTRY_ID.getFieldName())))
                .setTitle(cursor.getString(cursor.getColumnIndexOrThrow(TITLE.getFieldName())))
                .setDescription(cursor.getString(cursor.getColumnIndexOrThrow(DESCRIPTION.getFieldName())))
                .setCompleted(cursor.getInt(cursor.getColumnIndexOrThrow(COMPLETED.getFieldName())) > 0)
                .build();
    }

    @Override
    public ContentValues from(ContentValues contentValues, Task task) {
        contentValues.put(ENTRY_ID.getFieldName(), task.id());
        contentValues.put(TITLE.getFieldName(), task.title());
        contentValues.put(DESCRIPTION.getFieldName(), task.description());
        contentValues.put(COMPLETED.getFieldName(), task.completed());
        return contentValues;
    }
}

File: samples/advanced-samples/mvvm-sample/src/main/java/com/zhuinden/simplestackexamplemvvm/data/tasks/TaskTable.kt
package com.zhuinden.simplestackexamplemvvm.data.tasks


import android.content.ContentValues
import android.database.Cursor
import com.zhuinden.simplestackexamplemvvm.core.database.BaseTable
import com.zhuinden.simplestackexamplemvvm.core.database.DatabaseManager
import com.zhuinden.simplestackexamplemvvm.data.Task

/**
 * Created by Zhuinden on 2017.07.26..
 */
class TaskTable : BaseTable(), DatabaseManager.Mapper<Task> {
    override val fields: Array<DatabaseManager.Fields>
        get() = arrayOf(ENTRY_ID, TITLE, DESCRIPTION, COMPLETED)

    override fun getId(element: Any): String {
        return (element as Task).id!!
    }

    override val idFieldName: String
        get() = ENTRY_ID.fieldName

    override val tableName: String
        get() = TABLE_NAME

    override fun from(cursor: Cursor): Task {
        return Task(
            id = cursor.getString(cursor.getColumnIndexOrThrow(ENTRY_ID.fieldName)),
            title = cursor.getString(cursor.getColumnIndexOrThrow(TITLE.fieldName)),
            description = cursor.getString(cursor.getColumnIndexOrThrow(DESCRIPTION.fieldName)),
            completed = cursor.getInt(cursor.getColumnIndexOrThrow(COMPLETED.fieldName)) > 0
        )
    }

    override fun from(contentValues: ContentValues, task: Task): ContentValues {
        contentValues.put(ENTRY_ID.fieldName, task.id)
        contentValues.put(TITLE.fieldName, task.title)
        contentValues.put(DESCRIPTION.fieldName, task.description)
        contentValues.put(COMPLETED.fieldName, task.completed)
        return contentValues
    }

    companion object {
        const val TABLE_NAME = "task"
        val ENTRY_ID = DatabaseManager.Fields("entry_id", "text", "primary key")
        val TITLE = DatabaseManager.Fields("title", "text", "")
        val DESCRIPTION = DatabaseManager.Fields("description", "text", "")
        val COMPLETED = DatabaseManager.Fields("completed", "integer", "") // boolean

        val Q_ENTRY_ID = TABLE_NAME + "." + ENTRY_ID.fieldName
        val Q_TITLE = TABLE_NAME + "." + TITLE.fieldName
        val Q_DESCRIPTION = TABLE_NAME + "." + DESCRIPTION.fieldName
        val Q_COMPLETED = TABLE_NAME + "." + COMPLETED.fieldName
    }
}